kafka-client/src/main/java/dev/responsive/kafka/internal/utils/TableName.java

```diff
@@ -32,12 +32,12 @@ public class TableName {
       Pattern.compile("[^a-zA-Z0-9_]");
 
   private final String kafkaName;
-  private final String cassandraName;
+  private final String remoteName;
 
   public TableName(final String kafkaName) {
     this.kafkaName = kafkaName;
     final var escaped = kafkaName.replaceAll("_", "__"); // escape existing underscores
-    cassandraName = INVALID_CASSANDRA_CHARS
+    remoteName = INVALID_CASSANDRA_CHARS
```

Comment:
Out of curiosity, have you looked into the valid character set for mongo db names?

Also (and I'm not saying this is necessary the right thing to do) did you consider whether to enforce client-specific naming? I'm not SUPER worried about this, but for example there is a bug in the C* naming scheme since we flatter everything to lowercase whereas Streams will consider case when validating state store names. If MongoDB doesn't have the same constraints as C*, we could at least keep the bug contained to C*

Otherwise I think we can probably see all the state store names from the TopologyDescription, so we could parse it and enforce case-insensitive uniqueness of state store names ourselves

Reply:
yeah that was something I was going to do as a follow-up, we probably should be enforcing client specific naming but for now I just did it this way